src/uu/tload/src/tload.rs

```diff
@@ -0,0 +1,173 @@
+        })
+    }
+
+    #[cfg(not(target_os = "linux"))]
+    fn new() -> UResult<SystemLoadAvg> {
+        Ok(SystemLoadAvg::default())
+    }
+}
+
+#[allow(unused)]
+#[derive(Debug)]
+struct Settings {
+    delay: u64,
+    scale: usize, // Not used
+
+    is_modern: bool, // For modern display
+}
+
+impl Settings {
+    fn new(matches: &ArgMatches) -> Settings {
+        Settings {
+            delay: matches.get_one("delay").cloned().unwrap(),
+            scale: matches.get_one("scale").cloned().unwrap(),
+            is_modern: matches.get_flag("modern"),
+        }
+    }
+}
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+    let settings = Settings::new(&matches);
+
+    let mut terminal = ratatui::init();
+
+    let data = {
+        // Why 10240?
+        //
+        // Emm, maybe there will be some terminal can display more than 10000 char?
+        let data = Arc::new(RwLock::new(VecDeque::with_capacity(10240)));
+        data.write()
+            .unwrap()
+            .push_back(SystemLoadAvg::new().unwrap());
+        data
+    };
+    let cloned_data = data.clone();
+    thread::spawn(move || loop {
+        sleep(Duration::from_secs(settings.delay));
+
+        let mut data = cloned_data.write().unwrap();
+        if data.iter().len() >= 10240 {
+            // Keep this VecDeque smaller than 10240
+            data.pop_front();
+        }
+        data.push_back(SystemLoadAvg::new().unwrap());
+    });
+
+    loop {
+        // Now only accept `Ctrl+C` for compatible with GNU implementation
+        //
+        // Use `event::poll` for non-blocking event reading
+        if let Ok(true) = event::poll(Duration::from_millis(10)) {
+            // If event avaliable, break this loop
+            if let Ok(event::Event::Key(KeyEvent {
+                code: KeyCode::Char('c'),
+                modifiers: KeyModifiers::CONTROL,
+                ..
+            })) = event::read()
+            {
+                // GNU compatible
+                uucore::error::set_exit_code(130);
+                break;
+            }
+        }
+
+        terminal.draw(|frame| {
+            let data = &data.read().unwrap();
+            let data = data.iter().cloned().collect::<Vec<_>>();
+            frame.render_widget(
+                if settings.is_modern {
+                    ModernTui::new(&data)
+                } else {
+                    LegacyTui::new(&data)
+                },
+                frame.area(),
+            );
+        })?;
+
+        std::thread::sleep(Duration::from_millis(10));
+    }
+
+    ratatui::restore();
+    Ok(())
+}
+
+#[allow(clippy::cognitive_complexity)]
+pub fn uu_app() -> Command {
+    Command::new(uucore::util_name())
+        .version(crate_version!())
+        .about(ABOUT)
+        .override_usage(format_usage(USAGE))
+        .infer_long_args(true)
+        .disable_help_flag(true)
```

Comment:
What's the reason for disabling the help flag?

Reply:
oh, that's my fault